local-volume/provisioner/pkg/common/common.go

```diff
@@ -57,6 +64,8 @@ type MountConfig struct {
 	HostDir string `json:"hostDir"`
 	// The mount point of the hostpath volume
 	MountDir string `json:"mountDir"`
+	// The volumeType "block" or "file". If nil, the type defaults to "file"
+	VolumeType string `json:"volumeType"`
```

Comment:
@msau42 I actually went back and forth on this one. The question is what should we do if a discovery directory has a mix of block sym links and normal files? Should we expect users to keep a pristine discovery directory or else error out? By having an explicit declaration on what the directory should contain, we can discover what the user intended and ignore any entries that don't match that declaration. It seemed like a more forgiving process. Do let me know your thoughts on this, as I was on the fence about this.